cassandra/src/main/java/se/tre/freki/storage/cassandra/CassandraStore.java

```diff
@@ -426,28 +436,48 @@ public LabelId apply(final ResultSet rows) {
   @Nonnull
   ListenableFuture<List<String>> getNames(final LabelId id,
                                           final LabelType type) {
-    ResultSetFuture namesFuture = session.executeAsync(
-        getNameStatement.bind(toLong(id), type.toValue()));
+    ListenableFuture<List<Row>> namesFuture = getLabelRows(id, type);
 
-    return transform(namesFuture, new Function<ResultSet, List<String>>() {
+    return transform(namesFuture, new Function<List<Row>, List<String>>() {
       @Override
-      public List<String> apply(final ResultSet result) {
+      public List<String> apply(final List<Row> rows) {
         ImmutableList.Builder<String> builder = ImmutableList.builder();
 
-        for (final Row row : result) {
+        for (final Row row : rows) {
           final String name = row.getString("name");
           builder.add(name);
         }
-
         final ImmutableList<String> names = builder.build();
 
-        if (names.size() > 1) {
+        return names;
+      }
+    });
+  }
+
+  @Nonnull
+  ListenableFuture<List<Row>> getLabelRows(final LabelId id,
+                                           final LabelType type) {
+
+    ResultSetFuture namesFuture = session.executeAsync(
+        getNameStatement.bind(toLong(id), type.toValue()));
+
+    return transform(namesFuture, new Function<ResultSet, List<Row>>() {
+      @Override
+      public List<Row> apply(final ResultSet result) {
+        ImmutableList.Builder<Row> builder = ImmutableList.builder();
+
+        builder.addAll(result);
+
+        final ImmutableList<Row> rows = builder.build();
+
+        if (rows.size() > 1) {
           LOG.error("Found duplicate names for ({}, {})", id, type);
```

Comment:
Can we change the logging message to `Found duplicate ID to name mapping for ID {} with type {}`?